src/ui/RepoView.cpp

```diff
@@ -1963,7 +1963,15 @@ void RepoView::checkout(
     QPushButton *resetButton =
       dialog->addButton(tr("Reset Local Branch"), QMessageBox::AcceptRole);
     connect(resetButton, &QPushButton::clicked, [this, ref, local] {
-      createBranch(local, ref.target(), ref, true, true);
+      git::Branch head = mRepo.head();
+      if (head.isValid() && head.name() == local) {
+        // Reset HEAD branch.
```

Comment:
Add a comment to explain that this branch is currently checked out, so it can't be replaced.

Reply:
Done.